internal/cli/cmd/dac/setup/setup.go

```diff
@@ -0,0 +1,249 @@
+
+	_, err = io.Copy(outFile, response.Body)
+	if err != nil {
+		return fmt.Errorf("error copying content to file: %v", err)
+	}
+
+	logrus.Debug("Perses release archive downloaded successfully")
+
+	return nil
+}
+
+func extractCUEDepsToDst() error {
+	file, err := os.Open(archiveName)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Open the tar reader
+	gzipReader, err := gzip.NewReader(file)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+	tarReader := tar.NewReader(gzipReader)
+
+	// Extract each CUE dep to the destination path
+	// TODO simplify the code with https://github.com/mholt/archiver? Wait for stable release of v4 maybe
+	depsFolderFound := false
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		// Remove the wrapping folder for following evaluations
+		currentDepPath := removeFirstFolder(header.Name)
+
+		if currentDepPath == depsFolderName {
+			depsFolderFound = true
+		}
+		if !strings.HasPrefix(currentDepPath, depsFolderName) {
+			continue
+		}
+
+		newDepPath := fmt.Sprintf("%s/%s", depsRootDstPath, currentDepPath)
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(newDepPath, 0666); err != nil {
+				return fmt.Errorf("can't create dir %s: %v", newDepPath, err)
+			}
+			logrus.Debugf("dir %s created succesfully", newDepPath)
+		case tar.TypeReg:
+			outFile, err := os.Create(newDepPath)
+			if err != nil {
+				return fmt.Errorf("can't create file %s: %v", newDepPath, err)
+			}
+			defer outFile.Close()
+			if _, err := io.CopyN(outFile, tarReader, maxFileSize); err != nil {
+				return fmt.Errorf("can't copy content from %s: %v", header.Name, err)
+			}
+			logrus.Debugf("file %s extracted succesfully", newDepPath)
+		default:
+			return fmt.Errorf("unknown type: %b in %s", header.Typeflag, header.Name)
+		}
+	}
+
+	if !depsFolderFound {
+		return fmt.Errorf("CUE dependencies not found in archive")
+	}
+
+	return nil
+}
+
+func removeFirstFolder(filePath string) string {
+	separatorChar := "/" // force the usage of forward slash for strings comparison
+
+	// Split the path into individual components
+	components := strings.Split(filePath, separatorChar)
+
+	// Remove the top folder if there is at least one folder in the path
+	if len(components) > 1 {
+		components = components[1:]
+	}
+
+	// Join the components back into a path
+	resultPath := strings.Join(components, separatorChar)
+
+	return resultPath
+}
+
+func (o *option) SetWriter(writer io.Writer) {
+	o.writer = writer
+}
+
+func NewCMD() *cobra.Command {
+	o := &option{}
+	cmd := &cobra.Command{
+		Use:   "setup",
+		Short: "Sets up a local development environment to do Dashboard-as-Code",
+		Long: `
+takes care of adding the CUE sources from Perses as external dependencies to your DaC repo.
+/!\ It must be executed at the root of your repo.
+`,
+		Example: `
+# DaC setup when you are connected to a server
+percli dac setup
```

Comment:
you didn't like `percli dac init` ?

Reply:
actually that's because I couldnt name the package "init" since that's its reserved by the language 😅 myself I have no clear preference here, I can bring back `init` if you want and think about a good-enough package name..

Reply:
ah yeah I forgot that `init` is reserved.
Nethermind !